### ISSUE TITLE
I should be able to define my own error format

### ISSUE THREAD

**Issue by DeanPack** [zino-hofmann/graphql-flutter#1069]
**Describe the issue**
A clear and concise description of what the problem is.
I am creating a mobile app for an existing web app that uses GraphQL. The error responses are formatted simply as strings, and this package expects them to be a map of <String, dynamic> This cases the response parser to throw an error. I want to have a way to override the response parser.

**To Reproduce (MUST BE PROVIDED)**
1. recieve an error response that is formatted as a string and not a map
thats it

This code lives in a package you are using, the gql_link package. here is the snippet I'm talking about

```
  Response parseResponse(Map<String, dynamic> body) => Response(
        errors: (body["errors"] as List?)
            ?.map(
              (dynamic error) => parseError(error as Map<String, dynamic>),
            )
            .toList(),
        data: body["data"] as Map<String, dynamic>?,
        context: Context().withEntry(
          ResponseExtensions(
            body["extensions"],
          ),
        ),
      );
```
here is the path to that snippet

<img width="781" alt="Screen Shot 2022-03-15 at 11 25 13 PM" src="https://user-images.githubusercontent.com/32622006/158522384-95ab51f7-fa18-4f9f-b26a-489c87c1afe3.png">

**Expected behavior**

**device / execution context**
Are you on iOS, android, web, in a simulator, running from the terminal etc? This is especially important for `localhost` connection issues.

### Other useful/optional fields

Please fill or delete these sections if you don't fill them in

<details>
  <summary>Stacktrace:</summary>

```dart
{my stacktrace here}
```

</details>

**screenshots**

**additional context**
* What backend are you trying to use?
* If you have a network issue, does your operation work with [GraphiQL](https://www.electronjs.org/apps/graphiql)?

**additional notes**
the function says I can extend this to add non-standard behavior? I am not sure how to do that since this package doesn't provide any api's to extend the response parser (at least as I'm aware)

If you want to troubleshoot or discuss in real time, consider coming to the `support` channel in the [discord](https://discord.gg/tXTtBfC)

**Comment by DeanPack** [comment]
I typed out all of the fields, and when I clicked save, they were gone. Let me sum up my issue again:

graphql flutter uses another package, gql_link which handles the response parsing. In the response parser it expects the error to be Map<String,dynamic> and throws an exception if it isn't. In my case, when I have an error it is just a string. The responseparser says that it can be extended to override it, but I don't know how to do it since it is wrapped in this package and I don't see anything in the api's that can override the response parser.
here is the code snippet I'm talking about in gql_link
  /// Parses the response body
  ///
  /// Extend this to add non-standard behavior
  Response parseResponse(Map<String, dynamic> body) => Response(
        errors: (body["errors"] as List?)
            ?.map(
              (dynamic error) => parseError(error as Map<String, dynamic>),
            )
            .toList(),
        data: body["data"] as Map<String, dynamic>?,
        context: Context().withEntry(
          ResponseExtensions(
            body["extensions"],
          ),
        ),
      );
